sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -244,6 +244,71 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
       ("a", ClassData("a", 1)), ("b", ClassData("b", 2)), ("c", ClassData("c", 3)))
   }
 
+  test("REGEX column specification") {
+    val ds = Seq(("a", 1), ("b", 2), ("c", 3)).toDS()
+
+    intercept[AnalysisException] {
+      ds.select(expr("`(_1)?+.+`").as[Int])
+    }
+
+    intercept[AnalysisException] {
+      ds.select(expr("`(_1|_2)`").as[Int])
+    }
+
+    intercept[AnalysisException] {
+      ds.select(ds("`(_1)?+.+`"))
+    }
+
+    intercept[AnalysisException] {
+      ds.select(ds("`(_1|_2)`"))
+    }
```

Comment:
Could you capture the exception error messages? It can help reviewers ensure the error messages are correct. 

For example
```Scala
    val e = intercept[AnalysisException] { ds.select(expr("`(_1)?+.+`").as[Int]) }.getMessage
    assert(e.contains("xyz"))
```